automation/build.sh

```diff
@@ -25,99 +24,100 @@
 # to the docker registry.
 # If CLEANUP is "true", all images will be removed after pushing - including any relevant images
 # that may have been on the host from before the build, so be careful!
-# If ENABLE_TESTS is "true", tests will be run.
 #
-# Requires docker >= 17.05, and make.
+# Requires docker >= 17.05
 #
 
 set -e
 
 THIS_FILE=$0
 if [ -z "$ARCH" ] || [ -z "$TAG" ]; then
-	cat $THIS_FILE | awk '{if(/^#/)print;else exit}' | tail -n +2 | sed 's/\#//'
+	awk '{if(/^#/)print;else exit}' "${THIS_FILE}" | tail -n +2 | sed 's/\#//'
 	exit 1
 fi
 
-# This is the supervisor image we will produce
-TARGET_IMAGE=balena/$ARCH-supervisor:$TAG$DEBUG
+if ! [ -x "$(command -v npx)" ]; then
+  echo 'NPM/npx is required to execute this script' >&2
+  exit 1
+fi
 
-# Intermediate images and cache
-NODE_IMAGE=balena/$ARCH-supervisor-node:$TAG$DEBUG
-NODE_BUILD_IMAGE=balena/$ARCH-supervisor-node:$TAG-build$DEBUG
+# This is the supervisor image we will produce
+TARGET_IMAGE=balena/$ARCH-supervisor:$TAG
+TARGET_BUILD_IMAGE=balena/$ARCH-supervisor:$TAG-build
 
-TARGET_CACHE_MASTER=balena/$ARCH-supervisor:master$DEBUG
-NODE_CACHE_MASTER=balena/$ARCH-supervisor-node:master$DEBUG
-NODE_BUILD_CACHE_MASTER=balena/$ARCH-supervisor-node:master-build$DEBUG
+MASTER_IMAGE=balena/$ARCH-supervisor:master
+MASTER_BUILD_IMAGE=balena/$ARCH-supervisor:master-build
 
 CACHE_FROM=""
 function useCache() {
 	image=$1
 	# Always add the cache because we can't do it from
 	# a subshell and specifying a missing image is fine
-	CACHE_FROM="$CACHE_FROM --cache-from $image"
+	CACHE_FROM="--cache-from $image ${CACHE_FROM}"
```

Comment:
Any reason for the reordering?

Reply:
Was debugging a weird shell issue, I've reverted.